test/mocks/buffer/mocks.h

```diff
@@ -8,6 +8,49 @@
 #include "gmock/gmock.h"
 
 namespace Envoy {
+
+class MockBuffer : public Buffer::OwnedImpl {
+public:
+  MockBuffer() {
+    ON_CALL(*this, write(testing::_))
+        .WillByDefault(testing::Invoke(this, &MockBuffer::TrackWrites));
+    ON_CALL(*this, move(testing::_)).WillByDefault(testing::Invoke(this, &MockBuffer::BaseMove));
+  }
+
+  MOCK_METHOD1(write, int(int fd));
+  MOCK_METHOD1(move, void(Instance& rhs));
+  MOCK_METHOD2(move, void(Instance& rhs, uint64_t length));
+  MOCK_METHOD1(drain, void(uint64_t size));
+
+  void BaseMove(Instance& rhs) { Buffer::OwnedImpl::move(rhs); }
```

Comment:
Ironically I reread the style guide before sending this one out but Google3 style habits are so hard to break :-P